gobblin-runtime/src/test/java/org/apache/gobblin/util/SchedulerUtilsTest.java

```diff
@@ -263,12 +271,22 @@ public void onFileChange(Path path) {
       File newJobConfigFile = new File(this.subDir11, "test112.pull");
       Files.append("k1=v1", newJobConfigFile, ConfigurationKeys.DEFAULT_CHARSET_ENCODING);
 
-      semaphore.acquire(3);
-      Assert.assertEquals(fileAltered.size(), 3);
+      // Create file with same content again
+      File sameContentFile = new File(this.subDir3, "test31.PULL");
+      Properties jobProps5 = new Properties();
+      jobProps5.setProperty("k1", "b1");
+      // test-job-conf-dir/test3/test31.PULL
+      jobProps5.store(new FileWriter(sameContentFile), "");
+      // sleep for the observer to observe changes once
+      Thread.sleep(1500);
```

Comment:
Thread.sleep(1500L)

Reply:
consider using AssertWithBackoff instead of fix amount of sleep. that will improve the total runtime of the tests and help making timing dependent test more robust

Reply:
Done, thanks for informing about this.